donkeycar/vehicle.py

```diff
@@ -39,6 +42,9 @@ def add(self, part, inputs=[], outputs=[],
             run_condition: boolean
                 If a part should be run at all.
         """
+        assert type(inputs) is list, "inputs is not a list: %r" % inputs
+        assert type(outputs) is list, "outputs is not a list: %r" % outputs
+        assert type(threaded) is bool, "threaded is not a boolean: %r" % threaded
```

Comment:
Nice improvement.
With these assertions, the is no possibility, to have `inputs` and `outputs` different than `list`